src/Nethermind/Nethermind.Evm/EvmState.cs

```diff
@@ -44,45 +47,27 @@ public StackPool(int maxCallStackDepth = VirtualMachine.MaxCallDepth * 2)
             /// <param name="returnStack"></param>
             public void ReturnStacks(byte[] dataStack, int[] returnStack)
             {
-                _dataStackPool.Push(dataStack);
-                _returnStackPool.Push(returnStack);
+                _stackPool.Push(new(dataStack, returnStack));
             }
 
-            private byte[] RentDataStack()
-            {
-                if (_dataStackPool.TryPop(out byte[] result))
-                {
-                    return result;
-                }
-
-                _dataStackPoolDepth++;
-                if (_dataStackPoolDepth > _maxCallStackDepth)
-                {
-                    EvmStack.ThrowEvmStackOverflowException();
-                }
-
-                return new byte[(EvmStack.MaxStackSize + EvmStack.RegisterLength) * 32];
-            }
-
-            private int[] RentReturnStack()
+            public (byte[], int[]) RentStacks()
             {
-                if (_returnStackPool.TryPop(out int[] result))
+                if (_stackPool.TryPop(out StackItem result))
                 {
-                    return result;
+                    return (result.DataStack, result.ReturnStack);
```

Comment:
lets avoid returning new tuple here, return StackItem directly and add deconstructor to StackItem.